### PR TITLE
mcs: init HTTP handler

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

Need to merge #6960 first.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by rleungx** [comment]
PTAL @JmPotato @lhy1024

File: pkg/mcs/resourcemanager/server/server.go [hunk 1]
```diff
@@ -388,11 +395,8 @@ func (s *Server) startServer() (err error) {
 	if err != nil {
 		return err
 	}
-	s.listenURL, err = url.Parse(s.cfg.ListenAddr)
-	if err != nil {
-		return err
-	}
 	if tlsConfig != nil {
+		s.secure = true
```
Comment (lhy1024) [inline pkg/mcs/resourcemanager/server/server.go:399]: Do schedulering/server and tso/server need `s.secure = true`?

Comment (rleungx) [reply pkg/mcs/resourcemanager/server/server.go:399]: I thnk yes

Comment (lhy1024) [reply pkg/mcs/resourcemanager/server/server.go:399]: I don't find out `s.secure = true` in scheduling/server, is it in another pr?

**Review by lhy1024 — APPROVED** [review]

File: cmd/pd-server/main.go [hunk 2]
```diff
@@ -41,6 +41,11 @@ import (
 	"github.com/tikv/pd/server/config"
 	"github.com/tikv/pd/server/join"
 	"go.uber.org/zap"
+
+	// register microservice HTTP API
+	_ "github.com/tikv/pd/pkg/mcs/resourcemanager/server/apis/v1"
```
Comment (nolouch) [inline cmd/pd-server/main.go:46]: why need this? Does `install` pkg duplicate with it?

Comment (rleungx) [reply cmd/pd-server/main.go:46]: If we don't import, the independent service cannot use the HTTP interface in the current implementation. There is an import cycle problem.

Comment (nolouch) [reply cmd/pd-server/main.go:46]: I mean why not 
```
_ "github.com/tikv/pd/pkg/mcs/resourcemanager/server/install"
_ "github.com/tikv/pd/pkg/mcs/scheduling/server/install"
_ "github.com/tikv/pd/pkg/mcs/tso/server/apis/install"
```

make http and grpc register togeter. it exists import cycel? I think it's same with xxx/api.

Comment (rleungx) [reply cmd/pd-server/main.go]: fixed

**Review by JmPotato — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**